panels/updates/cc-updates-panel.c

```diff
@@ -0,0 +1,684 @@
+                                       on_automatic_updates_switch_changed_cb,
+                                       self);
+
+      gtk_switch_set_active (GTK_SWITCH (self->automatic_updates_switch), g_strcmp0 (value, "1") == 0);
+
+      g_signal_handlers_unblock_by_func (self->automatic_updates_switch,
+                                         on_automatic_updates_switch_changed_cb,
+                                         self);
+    }
+}
+
+static void
+load_tariff (CcUpdatesPanel *self)
+{
+  g_autoptr(MwtTariffLoader) tariff_loader = NULL;
+  g_autoptr(GBytes) bytes = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree gchar *tariff_path = NULL;
+  gchar *contents = NULL;
+  gsize size = 0;
+
+  g_debug ("Loading tariff");
+
+  tariff_path = g_build_filename (g_get_user_config_dir (), SYSTEM_TARIFF_FILENAME, NULL);
+  g_file_get_contents (tariff_path, &contents, &size, &error);
+  if (error)
+    {
+      if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+        g_warning ("Error reading tariff file: %s", error->message);
+      return;
+    }
+
+  bytes = g_bytes_new_take (contents, size);
+  tariff_loader = mwt_tariff_loader_new ();
+  mwt_tariff_loader_load_from_bytes (tariff_loader, bytes, &error);
+  if (error)
+    {
+      g_warning ("Error loading tariff: %s", error->message);
+      return;
+    }
+
+  g_signal_handlers_block_by_func (self->tariff_editor, on_tariff_changed_cb, self);
+  g_signal_handlers_block_by_func (self->scheduled_updates_switch, on_scheduled_updates_switch_changed_cb, self);
+
+  cc_tariff_editor_load_tariff (self->tariff_editor,
+                                mwt_tariff_loader_get_tariff (tariff_loader),
+                                &error);
+
+  if (error)
+    g_warning ("Error loading tariff: %s", error->message);
+  else
+    gtk_switch_set_active (GTK_SWITCH (self->scheduled_updates_switch), TRUE);
+
+  g_signal_handlers_unblock_by_func (self->scheduled_updates_switch, on_scheduled_updates_switch_changed_cb, self);
+  g_signal_handlers_unblock_by_func (self->tariff_editor, on_tariff_changed_cb, self);
+}
+
+static void
+save_tariff (CcUpdatesPanel *self)
+{
+  g_autoptr(GError) error = NULL;
+  g_autofree gchar *tariff_path = NULL;
+  GBytes *tariff_bytes;
+
+  tariff_path = g_build_filename (g_get_user_config_dir (), SYSTEM_TARIFF_FILENAME, NULL);
+  tariff_bytes = cc_tariff_editor_get_tariff_as_bytes (self->tariff_editor);
+
+  if (!tariff_bytes || !gtk_switch_get_active (GTK_SWITCH (self->scheduled_updates_switch)))
+    {
+      g_autoptr(GFile) file = g_file_new_for_path (tariff_path);
+
+      g_debug ("Deleting tariff");
+
+      g_file_delete_async (file,
+                           G_PRIORITY_LOW,
+                           NULL,
+                           (GAsyncReadyCallback) on_tariff_file_deleted_cb,
+                           &error);
+      return;
+    }
+
+  g_debug ("Saving tariff");
+
+  g_file_set_contents (tariff_path,
```

Comment:
If deleting the file is async, then setting the file’s contents should probably also be async.

Reply:
At this point, we're dealing with a tariff file that is exactly 110 bytes long. I'm afraid the added complexity of asynchronously saving the tariff surpasses the benefits.